torch/prototypes/nestedtensor/nested.py

```diff
@@ -0,0 +1,127 @@
+import torch
+
+
+def is_nested_tensor(obj):
+    return isinstance(obj, NestedTensor)
+
+
+def make_nested_tensor(obj):
+    if is_nested_tensor(obj):
+        return obj.clone().detach()
+    elif torch.is_tensor(obj):
+        return obj.clone().detach()
+    else:
+        assert isinstance(obj, list)
+        if len(obj) == 0:
+            return NestedTensor([])
+        for obj_ in obj:
+            assert(torch.is_tensor(obj_))
```

Comment:
You probably want a better error message in the future

Reply:
I'm in the process of replacing all asserts with Errors